Sources/MotorControlUI/MotionSensorStepView.swift

```diff
@@ -93,25 +171,47 @@ struct MotionSensorStepView: View {
                 guard !state.recorder.isPaused, countdown > 0 else { return }
                 countdown = max(countdown - 1, 0)
                 // Once the countdown hits zero, stop the recorder and *then* navigate forward.
-                // TODO: syoung 09/13/2022 Decide if this is causing weird stalling and refactor if needed.
                 if countdown == 0, state.recorder.status <= .running {
+                    audioFileSoundPlayer.vibrateDevice()
+                    speak(at: state.motionConfig.duration, completion: finishStep)
                     timer.upstream.connect().cancel()
-                    Task {
-                        do {
-                            state.result = try await state.recorder.stop()
-                        }
-                        catch {
-                            state.result = ErrorResultObject(identifier: state.node.identifier, error: error)
-                        }
-                        pagedNavigation.goForward()
-                    }
+                }
+                else {
+                    speak(at: clock.runningDuration())
                 }
             }
     }
+    
+    func speak(at timeInterval: TimeInterval, completion: (() -> Void)? = nil) {
+        guard let instruction = state.spokenInstructions[Int(timeInterval)],
+              !instructionCache.contains(instruction)
+        else {
+            completion?()
+            return
+        }
+        instructionCache.append(instruction)
+        state.voicePrompter.speak(text: instruction) { _, _ in
+            completion?()
+        }
+    }
+    
+    func finishStep() {
+        Task {
+            do {
+                state.result = try await state.recorder.stop()
+            }
+            catch {
+                state.result = ErrorResultObject(identifier: state.node.identifier, error: error)
+            }
+        }
+        pagedNavigation.goForward()
```

Comment:
Och! the call to `pagedNavigation.goForward()` needs to be inside the `Task { }` so that the step waits for the task to complete before finishing. Not sure how I messed that up. Sorry.